src/bson-encode.c

```diff
@@ -365,9 +365,10 @@ static void php_phongo_bson_append(bson_t* bson, php_phongo_field_path* field_pa
 				}
 
 				bson_append_array_begin(bson, key, key_len, &child);
-				field_path->current_level++;
+				php_phongo_field_path_write_type_at_current_level(field_path, PHONGO_FIELD_PATH_ITEM_ARRAY);
+				field_path->size++;
 				php_phongo_zval_to_bson_internal(entry, field_path, flags, &child, NULL TSRMLS_CC);
-				field_path->current_level--;
+				field_path->size--;
```

Comment:
Would it make sense to have macros or inline functions for incrementing and decrementing `size`?

Does `php_phongo_field_path_write_type_at_current_level()` ensure that there is always sufficient buffer space to increment `size`?

Reply:
Yes it does: https://github.com/mongodb/mongo-php-driver/pull/793/files#diff-f7d29a680148f52d6601f59ed787f577R147

Reply:
Yeah, it does. I can make it a macro, but as we're only doing this twice, I didn't see much use of it. Happy to do so if you would prefer it.

Reply:
I'm OK with keeping this as-is if it's only two uses.